--- maze0.c	2023-12-19 01:21:01.970532425 +0300
+++ maze1.c	2023-12-19 01:22:25.543123737 +0300
@@ -3,8 +3,14 @@
 #include <time.h>
 #include <string.h>
 
-int main() {
+int main(int argc, char **argv) {
 	int maze_size = 6;
+	if (argc > 1) {
+		int new_size = atoi(argv[1]);
+		if (new_size > 0) {
+			maze_size = new_size;
+		}
+	}
 	srand(time(NULL));
 	int *maze_sets = calloc(maze_size, sizeof(int));
 	char *r_walls = calloc(maze_size*maze_size, sizeof(char));
@@ -46,6 +52,7 @@
 			}
 		}
 		if (i == maze_size - 1) continue;
+
 		for (int j = 0; j < maze_size; j++) {
 			if (d_walls[i * maze_size + j] == 1) {
 				maze_sets[j] = num;
@@ -62,7 +69,7 @@
 			}
 		}
 	}
-	char *maze_r = calloc(2 * maze_size + 2, sizeof(char));
+	char *maze_r = calloc(2 * maze_size + 2, sizeof(char));	
 	memset(maze_r, '#', 2 * maze_size + 1);
 	maze_r[2 * maze_size + 1] = '\0';
 	printf("%s\n", maze_r);
